custom_components/tuya_local/devices/elegrp_dtr10_dimmer_switch.yaml

```diff
@@ -0,0 +1,193 @@
+        optional: true
+
+  # Bulb configuration
+  - entity: select
+    name: Bulb type
+    icon: "mdi:lightbulb-variant"
+    category: config
+    dps:
+      - id: 4
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: "Cfl"
+            value: CFL
+          - dps_val: "Led"
+            value: LED
+          - dps_val: "Incandescent"
+            value: Incandescent
+          - dps_val: "Halogen"
+            value: Halogen
+
+  # Settings
+  - entity: select
+    name: 'Long press "On"'
+    category: config
+    icon: "mdi:gesture-tap-hold"
+    dps:
+      - id: 108
+        type: integer
+        name: option
+        optional: true
+        mapping:
+          - dps_val: 258
+            value: "Nap 25%"
+          - dps_val: 505
+            value: "Cuddle 50%"
+          - dps_val: 753
+            value: "Leisure 75%"
+          - dps_val: 1000
+            value: "Bright 100%"
+  - entity: select
+    name: Fade on speed
+    icon: "mdi:transition"
+    category: config
+    dps:
+      - id: 103
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: "Immediate"
+            value: Immediate
+          - dps_val: "Fast"
+            value: Fast (1s)
+          - dps_val: "Medium"
+            value: Medium (5s)
+          - dps_val: "Slow"
+            value: Slow (15s)
+  - entity: select
+    name: Fade off speed
+    icon: "mdi:transition"
+    category: config
+    dps:
+      - id: 104
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: "Immediate"
+            value: Immediate
+          - dps_val: "Fast"
+            value: Fast (1s)
+          - dps_val: "Medium"
+            value: Medium (5s)
+          - dps_val: "Slow"
+            value: Slow (15s)
+  # Indicator LED converted to light (brightness only)
+  - entity: light
+    name: Night indicator light
+    category: config
+    icon: "mdi:led-on"
+    dps:
+      - id: 101
+        type: integer
+        name: brightness
+        unit: "%"
+        range:
+          min: 0
+          max: 100
+
+  # Brightness configuration
+  - entity: number
+    name: Minimum brightness
+    category: config
+    icon: "mdi:brightness-4"
+    dps:
+      - id: 3
+        type: integer
+        name: value
+        unit: "%"
+        range:
+          min: 0
+          max: 500
+        optional: true
+        mapping:
+          - scale: 10
+  # Used by the app to tell the light that the user is attempting to set the
+  # minimum brightness in the UI. Not useful for HA so we'll ignore it.
+  # - entity: switch
+  #   name: Minimum brightness enabled
+  #   category: config
+  #   icon: "mdi:toggle-switch"
+  #   dps:
+  #     - id: 114
+  #       type: boolean
+  #       name: switch
+  #       optional: true
+  # Value can't be changed. It seems like it is hardcoded to 1000 (100%) by the
```

Comment:
When value cannot be changed directly, sometimes these are affected by the light type setting (mainly minimum brightness). But since HA does not handle minimum brightness, it will not make any difference to include these even as read-only.